src/Microsoft.Health.SqlServer/WorkloadIdentityAcessTokenHandler.cs

```diff
@@ -0,0 +1,27 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core;
+using Azure.Identity;
+using Microsoft.Health.SqlServer.Configs;
+
+namespace Microsoft.Health.SqlServer;
+public class WorkloadIdentityAcessTokenHandler : IAccessTokenHandler
```

Comment:
Do we need a whole new handler for this? Could we instead use something like a [`ChainedTokenCredential`](https://learn.microsoft.com/en-us/dotnet/api/azure.identity.chainedtokencredential?view=azure-dotnet) to combine both `ManagedIdentityCredential` and `WorkloadIdentityCredential`?

Reply:
Yes that is possible. The reason why i added a new tokenhandler is because, i didnot wanted to change the original approach of using azureservicetokenprovider to fetch the new token. I have also added a new authentication type called worklaod identity. That way i am containing the affect the change i am making will have. For example, this change will  only take in effect if authentication is worklaodidentity. I am also planning to run few tests before on tip and low speed regions before enabling in all regions, Since we already have workload identity enabled in all regions, i found this approach will allow me to take control on switching back and forth between managed identity and workload identity.

Reply:
That makes sense, and I see in the [KEDA codebase](https://github.com/kedacore/keda/blob/0a1f7a3c46cf083e3609942302b2521d9c1b7949/pkg/scalers/azure/azure_azidentity_chain.go#L24) they express a concern around identity mismatches between the Pod Identity and the Workload Identity, so perhaps you're right in that it's best to be explicit.

However, I think we can instead of creating `IAccessTokenHandler` classes, can we change the existing `ManagedIdentityAccessTokenHandler` class to be more like an `AzureCredentialTokenHandler`? Or we could perhaps even get rid of the interface altogether and pass an `AzureCredential` directly to the `ManagedIdentitySqlConnectionBuilder`? That'd probably be the simplest solution.

Reply:
Addressed the comments . Please take a look